pkg/operator/ceph/cluster/controller.go

```diff
@@ -388,8 +389,29 @@ func (c *ClusterController) requestClusterDelete(cluster *cephv1.CephCluster) (r
 		return reconcile.Result{}, nil // do not requeue the delete
 	}
 
-	logger.Infof("cleaning up CephCluster %q", nsName)
+	logger.Info("removing networkFence if matching cephCluster UID found")
+	networkFenceList := &addonsv1alpha1.NetworkFenceList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		networkFenceLabel: string(cluster.GetUID()),
+	})
+	err := c.client.List(c.OpManagerCtx, networkFenceList, &client.MatchingLabelsSelector{Selector: labelSelector})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return reconcile.Result{}, errors.Wrap(err, "failed to list networkFence")
+	}
+	if len(networkFenceList.Items) > 0 {
+		for i := range networkFenceList.Items {
+			err = opcontroller.RemoveFinalizerWithName(c.OpManagerCtx, c.client, &networkFenceList.Items[i], "csiaddons.openshift.io/network-fence")
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to remove finalizer")
+			}
+		}
+		err := c.client.DeleteAllOf(c.OpManagerCtx, &addonsv1alpha1.NetworkFence{}, &client.MatchingLabelsSelector{Selector: labelSelector})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to delete networkFence with label %s", networkFenceLabel)
+		}
```

Comment:
```suggestion
		err := c.client.DeleteAllOf(c.OpManagerCtx, &addonsv1alpha1.NetworkFence{}, &client.MatchingLabelsSelector{Selector: labelSelector})
		if err != nil && !kerrors.IsNotFound(err) {
			return reconcile.Result{}, errors.Wrapf(err, "failed to delete networkFence with label %s", networkFenceLabel)
		}
```

@travisn I tried these to delete all the networkFence but it is not working as expected. NetworkFence was not deleted with the above code. Any suggestion?

Reply:
Nevermind, Madhu suggested the fix in the call.